cmd/kubelet/app/server.go

```diff
@@ -323,7 +324,7 @@ func startKubelet(k *kubelet.Kubelet, podCfg *config.PodConfig, kc *KubeletConfi
 
 func makePodSourceConfig(kc *KubeletConfig) *config.PodConfig {
 	// source of all configuration
-	cfg := config.NewPodConfig(config.PodConfigNotificationSnapshotAndUpdates)
+	cfg := config.NewPodConfig(config.PodConfigNotificationSnapshotAndUpdates, kc.EventSource)
```

Comment:
I think everything is a bit simpler if you pass around a record.EventRecorder instead of an api.EventSource?